rmf_task/include/rmf_task/agv/TaskPlanner.hpp

```diff
@@ -95,7 +94,14 @@ class TaskPlanner
     std::shared_ptr<rmf_traffic::agv::Planner> planner() const;
 
     /// Set the planner
-    Configuration& planner(std::shared_ptr<rmf_traffic::agv::Planner>);
+    Configuration& planner(std::shared_ptr<rmf_traffic::agv::Planner> planner);
+
+    /// Get the CostCalculator
+    std::shared_ptr<rmf_task::CostCalculator> cost_calculator() const;
```

Comment:
```suggestion
    const std::shared_ptr<rmf_task::CostCalculator>& cost_calculator() const;
```

We can also consider changing the other accessor functions to return their `std::shared_ptr` by const-reference.

Reply:
Done https://github.com/osrf/rmf_core/pull/278/commits/814174cffd80a473f649c39a5d3024e9fc7ce8bb